### PR TITLE
tpm2: add a test for PolicyPCR

### PR DESCRIPTION
**Pull request by awly**
Also, clarify what the expectDigest argument to PolicyPCR should be in
docs.

Updates #124

cc @karandave-bc

### CODE REVIEW

File: tpm2/tpm2.go [hunk 1]
```diff
@@ -781,6 +781,12 @@ func encodePolicyPCR(session tpmutil.Handle, expectedDigest tpmutil.U16Bytes, se
 }
 
 // PolicyPCR sets PCR state binding for authorization on a session.
+//
+// Note that expectedDigest must be a *digest* of the expected PCR value. You
```
Comment (josephlr) [inline tpm2/tpm2.go]: We should note here that expected digest is optional. If you provide it, the digest is checked, if you don't provide it, the call will always succeed (as if you had passed the right digest).

File: tpm2/test/tpm2_test.go [hunk 2]
```diff
@@ -1478,3 +1478,46 @@ func TestPlainImport(t *testing.T) {
 		t.Errorf("Got %X, expected %X", outBuff, inBuff)
 	}
 }
+
+func TestPolicyPCR(t *testing.T) {
```
Comment (josephlr) [inline tpm2/test/tpm2_test.go]: This is a great test. We should also add in a test for passing `nil` for the expected digest.

File: tpm2/tpm2.go (hunk 1 above)
Comment (awly) [reply tpm2/tpm2.go]: Done

File: tpm2/test/tpm2_test.go (hunk 2 above)
Comment (awly) [reply tpm2/test/tpm2_test.go]: Done

**Comment by awly** [comment]
PTAL

File: tpm2/test/tpm2_test.go [hunk 3]
```diff
@@ -1478,3 +1478,58 @@ func TestPlainImport(t *testing.T) {
...
+	t.Logf("expectedDigest=%x", expectedDigest)
+
+	tests := []struct {
+		desc           string
+		expectedDigest []byte
+		wantErr        bool
+	}{
+		// PolicyPCR expects a *digest* of the PCR value, not the value itself.
+		// Make sure PolicyPCR with value actually fails.
+		{desc: "success: digest passed in expectedDigest", expectedDigest: expectedDigest[:]},
```
Comment (josephlr) [inline tpm2/test/tpm2_test.go]: This is Good, but subtest descriptions are usually `CamelCase`, just like test names.

Comment (awly) [reply tpm2/test/tpm2_test.go]: Disagree, subtest descriptions are arbitrary text and `t.Run` normalizes them using few simple substitutions.
We've used camelCase in a few places here to match const/func names.

I'll merge without fixing this, but happy to discuss further offline.

Comment (josephlr) [reply tpm2/test/tpm2_test.go]: Np
